distributed/deploy/adaptive.py

```diff
@@ -223,7 +227,8 @@ def workers_to_close(self, **kwargs):
     @gen.coroutine
     def _retire_workers(self, workers=None):
         if workers is None:
-            workers = self.workers_to_close()
+            workers = self.workers_to_close(key=self.worker_key,
+                                            minimum=self.minimum)
```

Comment:
what is the logic for passing in minimum here? Should you also pass in maximum?

Reply:
Minimum here corresponds to the minimum number of workers that we want to keep.  The `workers_to_close` function is a slightly nicer place to handle this than outside.  For example if `workers_to_close` tells us that we need to close all of our workers, we then look at minimum and see that we need one, which worker should we keep?  Also, which *other* workers should we keep that are in the same group.  Suddenly we have to replicate the grouping logic.

Maximum doesn't make sense because we're only reducing the number of workers, not increasing them.